src/plugins/chart_expressions/expression_xy/common/types/expression_functions.ts

```diff
@@ -216,14 +315,43 @@ export interface LensMultiTable {
 export type ReferenceLineLayerConfigResult = ReferenceLineLayerArgs & {
   type: typeof REFERENCE_LINE_LAYER;
   layerType: typeof LayerTypes.REFERENCELINE;
+  table: Datatable;
+};
+
+export type ExtendedReferenceLineLayerConfigResult = ExtendedReferenceLineLayerArgs & {
+  type: typeof EXTENDED_REFERENCE_LINE_LAYER;
+  layerType: typeof LayerTypes.REFERENCELINE;
+  table: Datatable;
 };
 
-export type DataLayerConfigResult = DataLayerArgs & {
+export type DataLayerConfigResult = Omit<DataLayerArgs, 'palette'> & {
   type: typeof DATA_LAYER;
   layerType: typeof LayerTypes.DATA;
+  palette: PaletteOutput;
```

Comment:
Why omitting the `palette`, then adding the same type back?

Reply:
Because the palette is optional at the arguments, but it has `default` expression, which will surely produce the default palette.